### ISSUE TITLE
Question: Hillshade provider

### ISSUE THREAD

**Issue by Framstag** [janbar/osmin#9]
The screenshot suggests, this is already working. However the settings options does not seem to work. Bug or missing feature?

**Comment by janbar** [comment]
No missed. You have to create the file "hillshade-tile-server.json" (copying the provided sample file hillshade-tile-server.json.sample) in the app folder "resources". But if your USB is broken ...

**Comment by janbar** [comment]
The sample file provided with the app contains:
```
{
  "id": "wmflabs",
  "name": "wmflabs",
  "servers": [
    "http://tiles.wmflabs.org/hillshading/%1/%2/%3.png"
  ],
  "maximumZoomLevel": 18, 
  "copyright": "© wmflabs Hillshading"
}
```

**Comment by janbar** [comment]
I could create it by default on the first startup. But I don't know if we can use this server without limits. For myself I used my own hillshade server.

**Comment by Framstag** [comment]
Hmmm. The libosmscout answer would be to provide an offline solution. I do not know though if this is realistic. Would be worth to make a libosmscout feature request 😎

**Comment by janbar** [comment]
IMO not realistic using tiles from tiff. In my own hillshade server the cache of tiles data size 20gb and the tiff data more than 50gb. Make tiles cost a lot for the cpu using the basic method (gdal).

**Comment by Framstag** [comment]
I have now created https://github.com/Framstag/libosmscout/issues/1030 to evaluate the chances to get offline hillshading implemented.

**Comment by Framstag** [comment]
I also tried to find out what are the conditions for usage above tile server - but failed. Looks like it is part of the Wikipedia cloud infrastructure services: https://wikitech.wikimedia.org/wiki/Help:Cloud_Services_Introduction. The Teres and conditions do not explicitly forbid usage but frame it for tests.

**Comment by Karry** [comment]
Hi @janbar , you may use my hillshade server: `https://osmscout.karry.cz/hillshade/tile.php?z=%1&x=%2&y=%3`, its performance is low (when tile is not ready in cache), but it will survive few more requests :-) Or if you have some server that you can use, you may setup own, [here is howto](https://www.karry.cz/karry/blog/2019/01/08/hillshade_tile_server/).

**Comment by Framstag** [comment]
@janbar It would be nice to have the option to simply enable/disable @Karry hillshade server.

**Comment by janbar** [comment]
@Framstag , there is the option in settings. So I could copy the sample in the resource on startup if no one exists.

**Comment by Framstag** [comment]
For me this is OK, @Karry should keep an eye on it infrastructure though, since he may get some additional load (but I assume you did not do any advertising up to now or have an sign for significant usage?).

**Comment by janbar** [comment]
@Karry, @Framstag  , I configured the app to Karry server and I get many "too many request". It is a hard job to make these tiles and also requires throughput. Staying over Greece since 5 minutes the tiles doesn't download yet. I am afraid to break your infra. May be we think about later. For instance the sample file is plugged to http://tiles.wmflabs.org/hillshading/ , and you will have at least the Europe with a good precision.

**Comment by janbar** [comment]
The commit is fd2504ee87fb1dffd33b8a1801f8ad1f68505455.
The next release will be 1.3.4.
@Framstag , you will have to go in Settings / Configure Maps, then check the option Hillshade.

**Comment by janbar** [comment]
@Framstag , enabling hillshade will download tiles not yet cached in the phone from the internet. If you haven't wifi the phone will use the data mobil network.

**Comment by Framstag** [comment]
Works. You can  close the issue

**Comment by Karry** [comment]
> I get many "too many request".

There is limitation for two processes rendering tiles in parallel, because my raspberry has just 1 GiB of ram. Qt is using four parallel http streams (it is not configurable unfortunately), so when you are visiting area that is not cached, server have to reject some requests. But with some patience all tiles are rendered...
